mscore/prefsdialog.cpp

```diff
@@ -1172,13 +1234,15 @@ void PreferenceDialog::apply()
             preferences.setPreference(PREF_SCORE_STYLE_PARTSTYLEFILE, partStyle->text());
             MScore::defaultStyleForPartsHasChanged();
             }
-      
+
       WorkspacesManager::retranslateAll();
       preferences.setPreference(PREF_APP_WORKSPACE, WorkspacesManager::currentWorkspace()->name());
-      mscore->changeWorkspace(WorkspacesManager::currentWorkspace());
+      mscore->changeWorkspace(WorkspacesManager::currentWorkspace(), true); // with true you don't call preferencesChanged a second time (look at emit preferencesChanged())
+
       emit mscore->workspacesChanged();
-      
-      emit preferencesChanged();
+
+      emit preferencesChanged(false, uiStyleThemeChanged);
+      uiStyleThemeChanged = false;
```

Comment:
Same for the timer. The user should't be calling save really often, so why not use QTimer::singleshot with a nice lambda reenabling the button?